src/NuGet.Clients/PackageManagement.VisualStudio/EnvDTEProjectAdapter.cs

```diff
@@ -190,9 +192,41 @@ public NuGetFramework TargetNuGetFramework
         {
             get
             {
-                ThreadHelper.ThrowIfNotOnUIThread();
+                if (_nugetFramework != null)
+                {
+                    return _nugetFramework;
```

Comment:
Will re-targeting between RS2 and RS3 in the future reload the project? Is this safe to cache?